packages/expo-image-picker/android/src/main/java/expo/modules/imagepicker/ImagePickerOptions.kt

```diff
@@ -1,58 +1,76 @@
 package expo.modules.imagepicker
 
-import expo.modules.core.Promise
-import expo.modules.core.utilities.ifNull
-
-data class ImagePickerOptions(
-  val quality: Int,
-  val isAllowsEditing: Boolean,
-  val forceAspect: List<*>?,
-  val isBase64: Boolean,
-  val mediaTypes: MediaTypes,
-  val isExif: Boolean,
-  val videoMaxDuration: Int
-) {
-  companion object {
-    fun optionsFromMap(options: Map<String, Any?>, promise: Promise): ImagePickerOptions? {
-      val quality = options[ImagePickerConstants.OPTION_QUALITY]?.let {
-        if (it is Number) {
-          return@let (it.toDouble() * 100).toInt()
-        }
-
-        promise.reject(ImagePickerConstants.ERR_INVALID_OPTION, "Quality can not be `null`.")
-        return null
-      } ?: ImagePickerConstants.DEFAULT_QUALITY
+import android.content.Intent
+import androidx.annotation.FloatRange
+import androidx.annotation.IntRange
+import expo.modules.kotlin.assertions.assertValueGreaterOrEqual
+import expo.modules.kotlin.assertions.assertValueInRange
+import expo.modules.kotlin.exception.CodedException
+import expo.modules.kotlin.records.Field
+import expo.modules.kotlin.records.Record
 
-      val isAllowsEditing = options[ImagePickerConstants.OPTION_ALLOWS_EDITING] as? Boolean ?: false
+internal class ImagePickerOptions: Record {
+  @Field
+  @FloatRange(from = 0.0, to = 1.0)
+  var quality: Double = 0.2
+    set(value) {
+      assertValueInRange(value, lowerBound = 0.0, upperBound = 1.0)
+      field = value
+    }
 
-      val forceAspect: List<*>? = options[ImagePickerConstants.OPTION_ASPECT]?.let {
-        if (it is List<*> && it.size == 2 && it[0] is Number && it[1] is Number) {
-          return@let it
-        }
+  @Field
+  var allowsEditing: Boolean = false
 
-        promise.reject(ImagePickerConstants.ERR_INVALID_OPTION, "'Aspect option must be of form [Number, Number]")
-        return null
+  /**
+   * TODO(@bbarthec): undocumented
+   */
+  @Field
+  @IntRange(from = 0)
```

Comment:
@IntRange annotation isn't supported on Pair<Int, Int> type - it will lead to crash right now